### PR TITLE
Add RGB <-> HSV conversion functions for SKColor (#61)

### PR DESCRIPTION
**Pull request by tdenniston**
Issue #61.

This is a direct translation from the Skia functions `SkRGBToHSV` and `SkHSVToColor`. Happy to make any changes.

### CODE REVIEW

**Comment by migueldeicaza** [comment]
This looks good, but we should also expose properties like System.Drawing color does to get the individual HSV properties, very much like RGB exist today.

File: binding/Binding/Definitions.cs [hunk 1]
```diff
@@ -97,7 +97,123 @@ public SKColor WithAlpha (byte alpha)
...
+			case 1: r = q; g = v; b = p; break;
+			case 2: r = p; g = v; b = t; break;
+			case 3: r = p; g = q; b = v; break;
+			case 4: r = t; g = p; b = v; break;
+			default: r = v; g = p; b = q; break;
+			}
+			return new SKColor ((byte)r, (byte)g, (byte)b);
+		}
+
+		public static void ToHSV(SKColor color, out float hue, out float saturation, out float value)
```
Comment (migueldeicaza) [inline binding/Binding/Definitions.cs]: It is not clear why we would like to have this static method, as it requires an SKColor to be there.

I rather just keep the instance method.

**Comment by migueldeicaza** [comment]
See my comment about the static public API entry point.

Oddly, my other comment is gone.

We should also add properties to get the individual HSV properties, like we surface RGB.

**Comment by mattleibow** [comment]
Just want to say +1 for the unit test!

**Comment by tdenniston** [comment]
Thank you @xamarin-cla-bot ...

I fixed the issues from the above comments. I didn't know if we should cache the values of HSV in the SKColor instance or not. Currently they are computed lazily when you use the Hue, Saturation and Value property getters.

**Pull request closed without merging**